packages/pymedphys_dicom/src/pymedphys_dicom/dicom/dose.py

```diff
@@ -32,100 +30,148 @@
-
-    .. deprecated:: 0.5.0
-            `dose_from_dataset` will be removed in a future version of
-            PyMedPhys in favour of a new & improved API.
     """
 
     if set_transfer_syntax_uid:
         ds.file_meta.TransferSyntaxUID = pydicom.uid.ImplicitVRLittleEndian
 
     if reshape:
-        warnings.warn((
-            '`dose_from_dataset` currently reshapes the dose grid. In a '
-            'future version this will no longer occur. To begin using this '
-            'function without the reshape pass the parameter `reshape=False` '
-            'when calling `dose_from_dataset`.'), UserWarning)
-        pixels = np.transpose(
-            ds.pixel_array, (1, 2, 0))
-    else:
-        pixels = ds.pixel_array
+        warnings.warn(
+            ('The `reshape` parameter no longer does anything. Please remove '
+             'this parameter. In a future version this parameter will no '
+             'longer be accepted.'), UserWarning)
 
-    dose = pixels * ds.DoseGridScaling
+    dose = ds.pixel_array * ds.DoseGridScaling
 
     return dose
 
 
-def axes_and_dose_from_dicom(dicom_filepath):
-    ds = pydicom.dcmread(dicom_filepath, force=True)
-    axes = xyz_axes_from_dataset(ds)
-    dose = dose_from_dataset(ds)
+def dicom_dose_interpolate(interp_coords, dose: pydicom.Dataset):
+    """Interpolates across a DICOM dose dataset.
+
+    Parameters
+    ----------
+    dose : pydicom.Dataset
+        An RT DICOM Dose object
+    interp_coords : tuple(z, y, x)
+        A tuple of coordinates in DICOM order, z axis first, then y, then x
+        where x, y, and z are DICOM axes.
+    """
+
+    interp_z = np.array(interp_coords[0], copy=False)[:, None, None]
+    interp_y = np.array(interp_coords[1], copy=False)[None, :, None]
+    interp_x = np.array(interp_coords[2], copy=False)[None, None, :]
+
+    coords, dose = zyx_and_dose_from_dataset(dose)
+    interpolation = RegularGridInterpolator(coords, dose)
+
+    try:
+        result = interpolation((interp_z, interp_y, interp_x))
+    except ValueError:
+        print(f"coords: {coords}")
+        raise
 
-    return axes, dose
+    return result
 
 
-def load_dicom_data(ds, depth_adjust):
-    dose = dose_from_dataset(ds)
-    crossplane, vertical, inplane = xyz_axes_from_dataset(ds)
+def depth_dose(depths, dose: pydicom.Dataset, plan: pydicom.Dataset):
+    depths = np.array(depths, copy=False)
 
-    depth = vertical + depth_adjust
+    surface_entry_point = get_surface_entry_point(plan)
+    depth_adjust = surface_entry_point.y
 
-    return inplane, crossplane, depth, dose
+    y = depths + depth_adjust
+    x, z = [surface_entry_point.x], [surface_entry_point.z]
 
+    coords = (z, y, x)
+
+    extracted_dose = np.squeeze(dicom_dose_interpolate(coords, dose))
+
+    return extracted_dose
+
+
+def profile(displacements, depth, direction, dose: pydicom.Dataset,
+            plan: pydicom.Dataset):
+
+    displacements = np.array(displacements, copy=False)
```

Comment:
In its current form yes. It's essentially the output of the distance value within mephysto (called displacement because it felt funny having a negative distance)

Reply:
See https://github.com/pymedphys/pymedphys/pull/427#discussion_r306596494